internal/applier/helm.go

```diff
@@ -171,6 +164,10 @@ func (h *Helm) getReleaseState(cl helmclient.ActionInterface, ext *ocv1.ClusterE
 		}
 		return nil, desiredRelease, StateNeedsInstall, nil
 	}
+	if err != nil {
+		return nil, nil, StateError, err
```

Comment:
Should we really return as a StateError when the error is `driver.ErrReleaseNotFound` which means that the release was not found? Should not this scenario be return as `StateNeedsInstall` as described and in the original code?

What is the problem that you are facing?
Can you please describe What you did? What did you face, and what would you expect instead?

Reply:
`StateError` is not actually being returned for the case when the error is `driver.ErrReleaseNotFound`, only when it is not that specific error type and I am not changing current code behaviour in that respect.
The only thing this PR changes within the applier and `getReleaseState` is to remove the first `errors.Is(err, driver.ErrReleaseNotFound)` block and makes sure the second one runs (please see my response above).

Reply:
See; https://github.com/operator-framework/operator-controller/pull/1539#discussion_r1903235394